lib/PushManager.js

```diff
@@ -99,14 +94,23 @@ class PushManager {
 
         //this.socket.on('message',  function(queueKey, msg){logger.debug(`message received [${util.inspect(msg)}]`);});
         this.socket.on('message',  (...args) => this._messageReceived(...args));
+
+        this.socket.on('connected', (queueKey, queueKeyHash) =>
+        {
+            logger.debug('connected event [' + queueKey + '] hash [' + queueKeyHash + ']');
+            let deferred = this.registeredQueues[queueKeyHash].deferredObject;
+            this.registeredQueues[queueKey] = queueKeyHash;
+            deferred.resolve();
+        });
     }
 
     _messageReceived(queueKey, msg)
     {
         logger.debug(`message received [${util.inspect(msg)}]`);
         if (this.registeredQueues[queueKey])
         {
-            const objectId = this.registeredQueues[queueKey].objectId;
+            let queueKeyHash = this.registeredQueues[queueKey];
+            const objectId = this.registeredQueues[queueKeyHash].objectId;
```

Comment:
why sometimes registeredQueues saves hash and sometimes an object... it's confusing...
why not always use queueKey as the key?